### ISSUE TITLE
[Feature]: Transition/Animation Effects for Widgets

### ISSUE THREAD

**Issue by ame-appsmith** [appsmithorg/appsmith#15095]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

Add support for transition effects when displaying widgets. For example, when showing a modal or dynamically showing a hidden widget, the widget should be displayed using some types of PowerPoint transitions (e.g., showing a modal with a bouncing effect).

### Why should this be worked on?

Allowing users to configure transition effects for widgets would improve the user experience, as they could build more attractive apps.

**Comment by dilippitchika** [comment]
@ame-appsmith was this requested by a user?

**Comment by ame-appsmith** [comment]
@dilippitchika Yes, a [user on Intercom](https://app.intercom.com/a/apps/y10e7138/inbox/inbox/conversation/164629100138189#part_id=initial-part-164629100138189-1362421236) requested this feature.

**Comment by SaschaValentin** [comment]
Would be very interesting e.g. for building burger menu with sliding containers.

**Comment by dilippitchika** [comment]
@SaschaValentin are you talking about a drawer type UI, where it slides and opens a menu?

@ichik just an fyi

**Comment by SaschaValentin** [comment]
Yes, something like this. Could be a container where you place elements in. Maybe also a modal does the job. But I think we need some kind of animation (slide-in, fade-in etc.).
